Boto3 typing has been fixed and so CI no longer passes

The type stubs for Boto3 have been updated, fixing some bugs we were working around. This makes the workarounds fail MyPy.

```
/builds/databiosphere/toil/contrib/admin/mypy-with-ignore.py
src/toil/lib/aws/utils.py:252: error: Unused "type: ignore" comment
            info = s3_client.head_bucket(Bucket=bucket_name)  # type: igno...
            ^
src/toil/lib/aws/utils.py: note: In function "get_bucket_region":
src/toil/lib/aws/utils.py:253:16: error: Redundant cast to "str" 
[redundant-cast]
            return cast(str, info['ResponseMetadata']['HTTPHeaders']['x-am...
                   ^
Found 2 errors in 1 file (checked 59 source files)
```

We should fix the MyPy type checking, and pin versions of the stubs so this can't happen again to formerly-working commits.

--- a/requirements-aws.txt
+++ b/requirements-aws.txt
@@ -1,6 +1,6 @@
 boto>=2.48.0, <3
 boto3>=1.20.46, <2
-boto3-stubs[s3,sdb,iam]>=1.20.46.post1, <2
-mypy-boto3-s3>=1.20.46.post1, <2
-mypy-boto3-sdb>=1.20.46.post1, <2
-mypy-boto3-iam>=1.20.46.post1, <2
+boto3-stubs[s3,sdb,iam]==1.24.0
+mypy-boto3-s3==1.24.0
+mypy-boto3-sdb==1.24.0
+mypy-boto3-iam==1.24.0

--- a/src/toil/lib/aws/utils.py
+++ b/src/toil/lib/aws/utils.py
@@ -248,9 +248,8 @@ def get_bucket_region(bucket_name: str, endpoint_url: Optional[str] = None, only
         # for an 'x-amz-bucket-region' header on the response, which can tell
         # us where the bucket is. See
         # <https://github.com/aws/aws-sdk-cpp/issues/844#issuecomment-383747871>
-        # This won't typecheck until https://github.com/youtype/mypy_boto3_builder/issues/147 is fixed.
-        info = s3_client.head_bucket(Bucket=bucket_name)  # type: ignore
-        return cast(str, info['ResponseMetadata']['HTTPHeaders']['x-amz-bucket-region'])
+        info = s3_client.head_bucket(Bucket=bucket_name)
+        return info['ResponseMetadata']['HTTPHeaders']['x-amz-bucket-region']
 
     # Compose a list of strategies we want to try in order, which may work.
     # None is an acceptable return type that actually means something.

